knowledge-base/implementation-documentation/0005-api-improvements.md

```diff
@@ -214,6 +214,54 @@ Cons:
 - Difficult to implement.
 - Not realistic for 2.0, but maybe later.
 
+## Function style filtering [Updated]
+
+The new filter api is implemented as non-breaking change and the current methods are deprecated. Lambda filters on collection properties will be implemented as a new feature in the major version.
+
+### Current API
+
+```ts
+peopleApi.requestBuilder()
+  .getAll()
+  .filter(
+    peopleApi.schema.USER_NAME.equals('Peter'),
+    peopleApi.schema.PHOTOS.filter(photoApi.schema.NAME.equals('Photo1'))
+    peopleApi.schema.TO_FRIENDS.filter(
+      any(
+        peopleApi.schema.FIRST_NAME.equals('John'),
+        peopleApi.schema.LAST_NAME.equals('Miller')
+      )
+    )
+  );
+```
+
+### Examples
+
+**Option 1: Similar to the current api**
+
+```ts
+.filter(({ USER_NAME, PHOTOS, TO_FRIENDS, EMAILS, ADDRESSES }) => and(
+  USER_NAME.equals('Peter'),
+  PHOTOS.filter((photo) => photo.NAME.equals('Photo1')),
+  TO_FRIENDS.filter((friend) => any(friend.FIRST_NAME.equals('John'))),
+  EMAILS.filter(e => any(e.equals('peter@example.com'))) // Collection
+));
+```
+Issue remains: The inner `.filter` is confusing, because it implies that we are filtering the Friends/Photos/Emails here, while it is actually filtering people.
+
+**Option 2: Without inner `.filter` on properties**
+
+```ts
+.filter(({ USER_NAME, PHOTOS, TO_FRIENDS, EMAILS, ADDRESSES }) => and(
+    USER_NAME.equals('Peter'),
+    // PHOTOS.filter(({ NAME }) => NAME.equals('Photo1'))
```

Comment:
[q] I guess this is a "one to one" navigation property or complex type?
Maybe having examples for the types mentioned above would be helpful?

Reply:
This is a 1-to-1 nav property, I tried to add examples for all property types. Will add comments to make it more clear